pkg/webhook/injection.go

```diff
@@ -63,23 +86,65 @@ func (si *SidecarInjector) supportsNativeSidecar() (bool, error) {
 	return supportsNativeSidecar, nil
 }
 
-func injectSidecarContainer(pod *corev1.Pod, config *Config, supportsNativeSidecar bool) {
-	nativeSidecarEnabled := true
-	if enable, ok := pod.Annotations[GcsFuseNativeSidecarEnableAnnotation]; ok {
-		parsedAnnotation, err := ParseBool(enable)
-		if err != nil {
-			klog.Errorf("failed to parse enableNativeSidecar annotation... ignoring annotation: %v", err)
-		} else {
-			nativeSidecarEnabled = parsedAnnotation
-		}
-	}
-	if supportsNativeSidecar && nativeSidecarEnabled {
+func injectSidecarContainer(pod *corev1.Pod, config *Config, injectAsNativeSidecar bool) {
+	if injectAsNativeSidecar {
 		pod.Spec.InitContainers = insert(pod.Spec.InitContainers, GetNativeSidecarContainerSpec(config), getInjectIndex(pod.Spec.InitContainers))
 	} else {
 		pod.Spec.Containers = insert(pod.Spec.Containers, GetSidecarContainerSpec(config), getInjectIndex(pod.Spec.Containers))
 	}
 }
 
+func (si *SidecarInjector) injectMetadataPrefetchSidecarContainer(pod *corev1.Pod, config *Config, injectAsNativeSidecar bool) {
+	var containerSpec corev1.Container
+	var index int
+
+	injected, _ := validatePodHasSidecarContainerInjected(MetadataPrefetchSidecarName, pod, []corev1.Volume{}, []corev1.VolumeMount{})
+	if injected {
+		klog.Infof("%s is already injected, skipping injection...", MetadataPrefetchSidecarName)
+
+		return
+	}
+
+	// Extract user provided metadata prefetch sidecar image.
+	userProvidedMetadataPrefetchSidecarImage, err := ExtractImageAndDeleteContainer(&pod.Spec, MetadataPrefetchSidecarName)
```

Comment:
please create a bug to update https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/cloud-storage-fuse-csi-driver#private-image with metadata sidecar (starting gke version ...)

Reply:
Done.